src/components/lib/crt/crt.c

```diff
@@ -758,13 +805,23 @@ crt_sinv_create(struct crt_sinv *sinv, char *name, struct crt_comp *server, stru
 		.s_fn_addr   = s_fn_addr
 	};
 
-	if (crt_ns_vas_shared(client, server))
-		sinv->sinv_cap = cos_sinv_alloc(cli, srv->comp_cap_shared, sinv->s_fn_addr, client->id);
-	else
-		sinv->sinv_cap = cos_sinv_alloc(cli, srv->comp_cap, sinv->s_fn_addr, client->id);
-
+	comp_s = (srv->comp_cap_shared) ? srv->comp_cap_shared : srv->comp_cap;
+	sinv->sinv_cap = cos_sinv_alloc(cli, comp_s, sinv->s_fn_addr, client->id);
 	assert(sinv->sinv_cap);
-	printc("sinv %s cap %ld\n", name, sinv->sinv_cap);
+
+	if (crt_ns_vas_shared(client, server) && s_altfn_addr && c_fast_callgate_addr) {
```

Comment:
@evanstella @gparmer Do we need some assertions here to prevent code executing if only some of the conditions are true?